### ISSUE TITLE
Spotify

### ISSUE THREAD

**Issue by QuantumTunnel** [Just-Some-Bots/MusicBot#46]
Wouldn't it be possible to add spotify functionality into this bot?
I think you might have to have a premium account though.
Am I right or completeely off base?

**Comment by srhinos** [comment]
Hi, so it won't be possible for a number of legal issues. Our bot right now isn't _technically_ legal but youtube doesn't care. 

Spotify however does care and our bot would be breaking the Developer ToS in several places, most importantly

> 3h. Audio-Specific Restrictions. Streaming via the Spotify Platform shall only be made available to subscribers to the Premium Spotify Service. Please note that the Widgets and Audio Preview Clips may be made available to Spotify users who do not subscribe to the Premium service. You shall only make Audio Preview Clips available for streaming in the territories in which the corresponding full length track is available.

We'd have to do checks to make sure everyone in that voice channel is a Spotify Premium member along with making sure the bot owner provided a Spotify Premium account. Also we'd have to do numerous other checks which really can't be feasibly done. 

tl;dr: No. Its not legal.

**Comment by bencooper222** [comment]
@SexualRhinoceros What if we implemented functionality that iterated through the Spotify playlist, grabbed the names of each song, and looked them up on YouTube? I don't believe that would violate the ToS.

**Comment by jayktaylor** [comment]
@bencooper222 It would not violate the ToS however it is very unlikely to be added as a feature. Spotify's web API requires a user to go through its [auth flow](https://developer.spotify.com/web-api/authorization-guide/) in order to [retrieve playlist information](https://developer.spotify.com/web-api/get-playlists-tracks/) (see required header fields on that page). This would be difficult to implement in a bot like this one.

There are free web services like [PlaylistBuddy](http://www.playlistbuddy.com/) that can convert Spotify playlists to YouTube ones which we encourage people to use if they want to do that.
